CMakeLists.txt

```diff
@@ -32,10 +40,22 @@ if(BUILD_TESTING)
   ament_lint_auto_find_test_dependencies()
 
   ament_add_gtest(test_basic test/test_basic.cpp)
+
+  add_library(toy_test_library SHARED test/toy_test_library.cpp)
+  ament_add_gtest(test_find_library test/test_find_library.cpp)
+  target_link_libraries(test_find_library ${PROJECT_NAME} toy_test_library)
+  set_tests_properties(test_find_library PROPERTIES
+      ENVIRONMENT
+        "_TOY_TEST_LIBRARY_DIR=$<TARGET_FILE_DIR:toy_test_library>;_TOY_TEST_LIBRARY=$<TARGET_FILE:toy_test_library>")
 endif()
 
 ament_package()
 
 install(
   DIRECTORY include/
   DESTINATION include)
+install(
+  TARGETS ${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
```

Comment:
This needs a `RUNTIME DESTINATION` for Windows.

Also the new symbols need to use visibility macros to control the visibility of them on Windows (see other packages for examples).

Reply:
Is there a way to dispatch CI for this package, esp. for Windows?

Reply:
No, the CI jobs on ci.ros2.org need to be triggered manually.

Reply:
Done.

Reply:
> Also the new symbols need to use visibility macros to control the visibility of them on Windows (see other packages for examples).

This doesn't seem to be tackled yet.

Reply:
D'oh, yup - forgot that! Will add in...

Reply:
Er... It seems like visibility control is another bit of copy-pasta among packages. Will copy here, but will file an issue somewhere about it.

Reply:
Done.